### DISCUSSION TITLE
meson did not generate the build.ninja file

### DISCUSSION THREAD

**Discussion by ahmedmahany** [su2code/SU2#2196] [Q&A]
**Compiling SU2-8.0.0 on CentOS 7.9**
*When trying to generate the build.ninja file using the below command, it generates a file named build.ninja~ and this file does not include the commands that compile the source code*
`./meson.py build`

**HPC**
 - OS: [e.g., Linux (CentOS 7.9)]
 - C++ compiler and version: (GCC) 4.8.5
 - Python version: 3.6.8
 - SU2 Version: v8.0.0

**Comment by bigfooted** [comment]
What exactly does not work for you?
What is the output of 
`./meson.py build`
and what is the output of 
`./ninja -C build install`

**Comment by ahmedmahany** [comment]
Thanks for your response. 

Here you are 

##### The output of ./meson.py build 

[user@server su2-8.0.0]$ ./meson.py build
The Meson build system
Version: 0.61.1
Source dir: /home/user/support/su2-8.0.0
Build dir: /home/user/support/su2-8.0.0/build
Build type: native build
Project name: SU2
Project version: 8.0.0 "Harrier"
C compiler for the host machine: /app/utils/gcc-11.3.0/bin/gcc (gcc 11.3.0 "gcc (GCC) 11.3.0")
C linker for the host machine: /app/utils/gcc-11.3.0/bin/gcc ld.bfd 2.27-44
C++ compiler for the host machine: c++ (gcc 11.3.0 "c++ (GCC) 11.3.0")
C++ linker for the host machine: c++ ld.bfd 2.27-44
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/app/utils/anaconda3-2022/bin/python3)
WARNING: Python files installed by Meson might not be found by python interpreter.
 This warning can be avoided by setting "python.platlibdir" option.
WARNING: Python files installed by Meson might not be found by python interpreter.
 This warning can be avoided by setting "python.purelibdir" option.
Compiler for C++ supports arguments -Wno-ignored-qualifiers: YES
Compiler for C++ supports arguments -Wno-unused-parameter: YES
Compiler for C++ supports arguments -Wno-deprecated-declarations: YES
Compiler for C++ supports arguments -Wno-error=cast-function-type: YES
Compiler for C++ supports arguments -Wno-cast-function-type: YES
Compiler for C++ supports arguments -Wno-error=stringop-truncation: YES
Compiler for C++ supports arguments -Wno-stringop-truncation: YES
Compiler for C++ supports arguments -Wno-error=missing-field-initializers: YES
Compiler for C++ supports arguments -Wno-missing-field-initializers: YES
Compiler for C++ supports arguments -Wno-error=class-memaccess: YES
Compiler for C++ supports arguments -Wno-class-memaccess: YES
Found pkg-config: /usr/bin/pkg-config (0.27.1)
mpicc found: NO
Run-time dependency MPI for c found: NO (tried pkgconfig, config-tool and system)
mpic++ found: NO
Run-time dependency MPI for cpp found: NO (tried pkgconfig, config-tool and system)
Compiler for C supports arguments -Wno-unused-result: YES
Compiler for C supports arguments -Wno-unused-parameter: YES
Compiler for C supports arguments -Wno-unused-variable: YES
Compiler for C supports arguments -Wno-unused-but-set-variable: YES
Compiler for C supports arguments -Wno-sign-compare: YES
Compiler for C supports arguments -Wno-error=unused-function: YES
Compiler for C supports arguments -Wno-pedantic: YES
Compiler for C supports arguments -Wno-error=stringop-truncation: YES
Compiler for C supports arguments -Wno-stringop-truncation: YES
Compiler for C supports arguments -Wno-error=absolute-value: YES
Compiler for C supports arguments -Wno-error=class-memaccess: YES
Compiler for C supports arguments -Wno-unused-result: YES (cached)
Compiler for C supports arguments -Wno-unused-parameter: YES (cached)
Compiler for C supports arguments -Wno-unused-variable: YES (cached)
Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
Compiler for C supports arguments -Wno-sign-compare: YES (cached)
Compiler for C supports arguments -Wno-stringop-overflow: YES
Compiler for C supports arguments -Wno-error=discarded-qualifiers: YES
Compiler for C supports arguments -Wno-error=maybe-uninitialized: YES
Compiler for C supports arguments -Wno-error=unused-function: YES (cached)
Compiler for C supports arguments -Wno-error=unused-label: YES
Compiler for C supports arguments -Wno-pedantic: YES (cached)
Compiler for C supports arguments -Wno-error=stringop-truncation: YES (cached)
Compiler for C supports arguments -Wno-error=cast-function-type: YES
Compiler for C supports arguments -Wno-stringop-truncation: YES (cached)
Run-time dependency zlib found: YES 1.2.7
Found CMake: /app/utils/cmake-3.18.4/bin/cmake (3.18.4)
Run-time dependency szip found: NO (tried pkgconfig and cmake)
Has header "zlib.h" : YES
Checking for function "inflate" with dependency zlib: YES
Has header "sys/file.h" : YES
Has header "sys/ioctl.h" : YES
Has header "sys/resource.h" : YES
Has header "sys/socket.h" : YES
Has header "sys/stat.h" : YES
Has header "sys/time.h" : YES
Has header "sys/types.h" : YES
Has header "features.h" : YES
Has header "dirent.h" : YES
Has header "setjmp.h" : YES
Has header "stddef.h" : YES
Has header "stdint.h" : YES
Has header "unistd.h" : YES
Has header "io.h" : NO
Has header "winsock2.h" : NO
Has header "globus/common.h" : NO
Has header "pdb.h" : NO
Has header "pthread.h" : YES
Has header "srbclient.h" : NO
Has header "string.h" : YES
Has header "strings.h" : YES
Has header "stdlib.h" : YES
Has header "memory.h" : YES
Has header "dlfcn.h" : YES
Has header "inttypes.h" : YES
Has header "netinet/in.h" : YES
Has header "netdb.h" : YES
Has header "arpa/inet.h" : YES
Has header "stdbool.h" : YES
Has header "quadmath.h" : YES
Library m found: YES
Checking for function "ceil" with dependency -lm: YES
Library dl found: YES
Checking for function "dlopen" with dependency -ldl: YES
Library ucb found: NO
Checking for function "gethostbyname" with dependency -lucb: NO
Checking for function "lseek64" : NO
Checking for function "fseeko" : NO
Checking for function "stat64" : NO
Checking for size of "char" : 1
Checking for size of "short" : 2
Checking for size of "int" : 4
Checking for size of "unsigned" : 4
Checking for size of "long" : 8
Checking for size of "long long" : 8
Checking for size of "__int64" : -1
Checking for size of "float" : 4
Checking for size of "double" : 8
Checking for size of "long double" : 16
Checking for size of "int8_t" : 1
Checking for size of "uint8_t" : 1
Checking for size of "int_least8_t" : 1
Checking for size of "uint_least8_t" : 1
Checking for size of "int_fast8_t" : 1
Checking for size of "uint_fast8_t" : 1
Checking for size of "int16_t" : 2
Checking for size of "uint16_t" : 2
Checking for size of "int_least16_t" : 2
Checking for size of "uint_least16_t" : 2
Checking for size of "int_fast16_t" : 8
Checking for size of "uint_fast16_t" : 8
Checking for size of "int32_t" : 4
Checking for size of "uint32_t" : 4
Checking for size of "int_least32_t" : 4
Checking for size of "uint_least32_t" : 4
Checking for size of "int_fast32_t" : 8
Checking for size of "uint_fast32_t" : 8
Checking for size of "int64_t" : 8
Checking for size of "uint64_t" : 8
Checking for size of "int_least64_t" : 8
Checking for size of "uint_least64_t" : 8
Checking for size of "int_fast64_t" : 8
Checking for size of "uint_fast64_t" : 8
Checking for size of "size_t" : 8
Checking for size of "ssize_t" : -1
Checking for size of "ptrdiff_t" : -1
Checking for size of "off_t" : 8
Checking for size of "off64_t" : -1
Checking for size of "time_t" : 8
Checking for size of "bool" : 1
Checking for size of "_Quad" : -1
Checking for size of "__float128" : 16
Checking if "check dev_t" : compiles: YES
Checking if "check dev_t scalar" runs: NO (1)
Checking if "CLOCK_MONOTONIC" : compiles: YES
Checking whether type "struct tm" has member "tm_gmtoff" : NO
Checking whether type "struct tm" has member "__tm_gmtoff" : YES
Checking whether type "struct tm" has member "tm_sec" : YES
Checking whether type "struct tm" has member "tz_minuteswest" : NO
Checking for function "gettimeofday" : YES
Checking whether type "struct stat" has member "st_blocks" : YES
Checking for function "ioctl" : YES
Header <sys/ioctl.h> has symbol "TIOCGWINSZ" : YES
Header <sys/ioctl.h> has symbol "TIOCGETD" : YES
Checking for function "getpwuid" : YES
Checking for function "alarm" : YES
Checking for function "fcntl" : YES
Checking for function "flock" : YES
Checking for function "fork" : YES
Checking for function "frexpf" : YES
Checking for function "frexpl" : YES
Checking for function "gethostname" : YES
Checking for function "getrusage" : YES
Checking for function "llround" : YES
Checking for function "llroundf" : YES
Checking for function "lround" : YES
Checking for function "lroundf" : YES
Checking for function "lstat" : YES
Checking for function "pread" : YES
Checking for function "pwrite" : YES
Checking for function "rand_r" : YES
Checking for function "random" : YES
Checking for function "round" : YES
Checking for function "roundf" : YES
Checking for function "setsysinfo" : NO
Checking for function "signal" : YES
Checking for function "longjmp" : YES
Checking for function "setjmp" : YES
Checking for function "siglongjmp" : YES
Checking for function "sigsetjmp" : NO
Checking for function "sigprocmask" : YES
Checking for function "snprintf" : YES
Checking for function "srandom" : YES
Checking for function "strdup" : YES
Checking for function "strtoll" : YES
Checking for function "strtoull" : YES
Checking for function "symlink" : YES
Checking for function "system" : YES
Checking for function "tmpfile" : YES
Checking for function "asprintf" : YES
Checking for function "vasprintf" : YES
Checking for function "waitpid" : YES
Checking for function "vsnprintf" : YES
Header <setjmp.h> has symbol "sigsetjmp" : NO
Checking if "inline keyword check" : compiles: YES
Run-time dependency threads found: YES
Configuring H5pubconf.h using configuration
Checking for size of "long" : 8
Checking for size of "void *" : 8
Configuring cgnstypes.h using configuration
Configuring cg_hash_types.h using configuration
Configuring cgnsconfig.h using configuration
WARNING: You should add the boolean check kwarg to the run_command call.
         It currently defaults to false,
         but it will default to true in future releases of meson.
         See also: https://github.com/mesonbuild/meson/issues/9300
Message: Boost sources found.
Compiler for C++ supports arguments -Wno-misleading-indentation: YES
Compiler for C++ supports arguments -Wno-uninitialized: YES
Compiler for C++ supports arguments -Wno-placement-new: YES
Compiler for C++ supports arguments -Wno-pedantic: YES
Compiler for C++ supports arguments -Wno-error=deprecated-copy: YES
Dependency threads found: YES unknown (cached)
Message: -------------------------------------------------------------------------
         |    ___ _   _ ___                                                      |
         |   / __| | | |_  )   Release 8.0.0 "Harrier"                           |
         |   \__ \ |_| |/ /                                                      |
         |   |___/\___//___|   Meson Configuration Summary                       |
         |                                                                       |
         -------------------------------------------------------------------------

         Option          Value
         ---------------------
         TecIO:          true
         CGNS:           true
         AD (reverse):   false
         AD (forward):   false
         Python Wrapper: false
         Intel-MKL:      false
         OpenBlas:       false
         PaStiX:         false
         Mixed Float:    false
         libROM:         false
         CoolProp:       false
         MLPCpp:         false

         Please be sure to add the $SU2_HOME and $SU2_RUN environment variables,
         and update your $PATH (and $PYTHONPATH if applicable) with $SU2_RUN

         Based on the input to this configuration, add these lines to your .bashrc file:

         export SU2_RUN=/usr/local/bin
         export SU2_HOME=/home/user/support/su2-8.0.0
         export PATH=$PATH:$SU2_RUN
         export PYTHONPATH=$PYTHONPATH:$SU2_RUN

         Use './ninja -C build install' to compile and install SU2

Build targets in project: 15

Found ninja-1.10.0.git at /home/user/support/su2-8.0.0/ninja
Generating targets:   0%|                                                    | 0/15 [00:00<?, Generating targets:  33%|██████████████▋                             | 5/15 [00:00<00:00, 23.7Generating targets:  53%|███████████████████████▍                    | 8/15 [00:00<00:00, 22.5Generating targets:  73%|███████████████████████████████▌           | 11/15 [00:00<00:00, 23.6                                                                                              

A full log can be found at /home/user/support/su2-8.0.0/build/meson-logs/meson-log.txt

##### The output of ./ninja -C build install 

[user@server su2-8.0.0]$ ./ninja -C build install
ninja: Entering directory `build'
ninja: error: loading 'build.ninja': No such file or directory


This error happens because there is no file named build.ninja in the build directory. However, the meson command generates a file named build.ninja~ and it contains the following:

'# This is the build file for project "SU2"
'# It is autogenerated by the Meson build system.
'# Do not edit by hand.

ninja_required_version = 1.8.2

**Reply by bigfooted** [reply]
Anything else in the meson-log.txt file?
Is that the only contents of build.ninja or does it also contain all the rules? It looks like meson opened the temporary file for writing the build info but never finished writing it. 
Did you use git clone or one of the zip files to download it?

**Reply by ahmedmahany** [reply]
The meson-log.txt file contains no additional information. I downloaded the code from the releases section released on August 29, 2023.

**Reply by ahmedmahany** [reply]
@bigfooted Could you please help me in resolving this issue? Thank you

**Reply by bigfooted** [reply]
There is maybe a package that you are missing in your distribution. But the main packages that you need should be downloaded automatically. Do you have meson and ninja in the externals subfolder?

**Reply by ahmedmahany** [reply]
Thanks for your reply. The root folder includes meason.py script where this script compiles ninja and generates its executable (ninja), but it did not generate its build file correctly.
